src/Tools/dotnet-dump/Dumper.cs

```diff
@@ -131,19 +132,16 @@ public int Collect(TextWriter stdOutput, TextWriter stdError, int processId, str
                     client.WriteDump(dumpType, output, flags);
                 }
             }
-            catch (Exception ex) when
-                (ex is FileNotFoundException or
-                 ArgumentException or
-                 DirectoryNotFoundException or
-                 UnauthorizedAccessException or
-                 PlatformNotSupportedException or
-                 UnsupportedCommandException or
-                 InvalidDataException or
-                 InvalidOperationException or
-                 NotSupportedException or
-                 DiagnosticsClientException)
+            catch (Exception ex)
```

Comment:
Why did we remove the exception filter here?  Seems like this could mask a real bug (like a nullref).

Reply:
It was changed to catch all exceptions and display the full exception text if --diag was passed.  Otherwise, the exception message is displayed.  Nothing should be masked.  All exceptions should be catch and displayed.